src/mon/MgrStatMonitor.cc

```diff
@@ -66,6 +66,77 @@ void MgrStatMonitor::create_initial()
   encode(service_map, pending_service_map_bl, CEPH_FEATURES_ALL);
 }
 
+void MgrStatMonitor::calc_pool_availability()
+{
+  dout(20) << __func__ << dendl;
+  for (const auto& i : digest.pool_pg_unavailable_map) {
+    const auto& poolid = i.first;
+    if (pool_availability.find(poolid) == pool_availability.end()){
```

Comment:
think we can do this:
before the for loop:
```
auto pool_avail_end = pool_availability.end();
```

```suggestion
    if (pool_availability.find(poolid) == pool_avail_end){
```

this basically will avoid us calling for end iterator everytime.